### DISCUSSION TITLE
More information about the formats urls that FlyLeaf lib accepts as inputs

### DISCUSSION THREAD

**Discussion by stefantripac** [SuRGeoNix/Flyleaf#475] [Q&A] [answered]
2 weeks ago, I try to use the one of the C sharp libraries for video playing and/or transcoding, I tested FlyLeaf Lib and LIbVLCSharp, I want to integrate a player into an existing application( framework4.8, wpf), sensitive features for me are also: accessible snapshots in memory and some transcoding features. LibVlcSharp has transcoding features but for snapshots, it only offers snapshots into files, but files are only big size because hardcoded .png  in video.c . The FlayLeaf lib has the "TakeSnapshotToBitmap" that match my wishes and also many types of formats that can be played, 
1. I want to know some explicit examples of the  format urls that I can pass to Open  with FlayLeaf Lib. For example I can add transcoding option from ffmpeg to the input url string ? , 
2. I found in the "Config.cs" of the Lib the commment: /// <para>When enabled you can pass urls in this format device://[device_name]?[FFmpeg_Url]</para>///, can some one  tell me what are option [FFmpeg_Url]  with some examples for size ,  video  , coding  output format ... ?
Thank you

**Comment by SuRGeoNix** [comment, marked as answer]
Hi @stefantripac, indeed flyleaf currently does not support transcoding as it was focusing on playback only (at least initially). The upcoming v4 will be platform independent and more general purpose media processing library (including transcoding + filtering). You can find some more info for the fmt urls in the comments here: https://github.com/SuRGeoNix/Flyleaf/blob/master/FlyleafLib/MediaFramework/MediaDemuxer/Demuxer.cs#L362

You cannot currently do the transcoding with just a Url. You could manually write a transcoder and pass the snapshot hardware frame for it. That will not be easy though.

**Reply by stefantripac** [reply]
Thank you for help, a really appreciate your effort, I'm waiting for upcoming v4.

Best regards

**Reply by stefantripac** [reply]
Hi, Is it possible to Open with FlyLeaf  a stream with  MPEG-TS format, if yes how?

Thank you

**Reply by SuRGeoNix** [reply]
Use Config.Demuxer.ForceFormat to specify the input format (in your case "mpegts")

**Reply by stefantripac** [reply]
Ok , it works ,

Thank you.

**Reply by stefantripac** [reply]
Hi, I red this comment in Config.cs :  { "rtsp_transport",     "tcp" },                                     // Seems UDP causing issues,

I tested with RTSP on UDP, it works, What are the issues that you warn ?

**Reply by SuRGeoNix** [reply]
When you lose a packet within a GOP you will have corrupted frames until the next one. Except that I think there was a 'bug' with D3D11 and hardware acceleration with UDP. Generally, TCP is consider more reliable.

**Reply by stefantripac** [reply]
Thank you for the answer.

**Reply by stefantripac** [reply]
Hi, I tested the Bitrate parameter value using two input commands:

first: "device://dshow?video=PC-LM1E&framerate=10"  the Bitrate = 147,456 Kbps

second : "device://dshow?video=PC-LM1E&framerate=30"  the Bitrate = 37,220 Kbps

I expected that when framerate is greater the Bitrate should be also greater. How to how to interpret such behavior ?

Thanks,

**Reply by stefantripac** [reply]
I found the explanation, the camera switch the codec type from rawvideo  when framerate=10  to mpeg when framerate=30.
